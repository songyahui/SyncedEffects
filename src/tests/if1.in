I2(6);
I1;
I1 I2(1);
