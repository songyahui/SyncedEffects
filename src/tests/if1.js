"use hopscript"

var hh = require("hiphop");

var inSig = {accessibility: hh.IN};

var prg =
    <hh.module I1=${inSig} O1 I2=${inSig} O2>
      <hh.loop>
	<hh.if apply=${function() {return this.I1.now}}>
	  <hh.emit O1/>
	</hh.if>
	<hh.if apply=${function() {return this.I2.nowval > 2}}>
	  <hh.emit O2/>
	</hh.if>
	<hh.pause/>
      </hh.loop>
    </hh.module>

exports.prg = new hh.ReactiveMachine(prg, "if1");
