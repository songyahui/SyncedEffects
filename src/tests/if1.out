if1> I2(6);
--- Output: O2
if1> I1;
--- Output: O1 O2
if1> I1 I2(1);
--- Output: O1
